src/Microsoft.OData.Edm/Csdl/Serialization/EdmModelCsdlSchemaWriter.cs

```diff
@@ -400,6 +401,31 @@ internal void WriteOperationParameterElementHeader(IEdmOperationParameter parame
             }
         }
 
+        internal void WriteOperationParameterEndElement(IEdmOperationParameter parameter)
+        {
+            IEdmOptionalParameter optionalParameter = parameter as IEdmOptionalParameter;
+            if (optionalParameter != null && !(optionalParameter.VocabularyAnnotations(this.model).Where(a => a.Term == CoreVocabularyModel.OptionalParameterTerm).Count() > 0))

```

Comment:
>Count() > 0 [](start = 163, length = 11)

.Any() potentially avoids enumerating the collection and provides the same answer.